pkg/printers/internalversion/printers.go

```diff
@@ -507,14 +507,20 @@ func formatEndpoints(endpoints *api.Endpoints, ports sets.String) string {
 	return ret
 }
 
-// translateTimestamp returns the elapsed time since timestamp in
+// translateElapsedTimeUntilNow returns the elapsed time since timestamp in
 // human-readable approximation.
-func translateTimestamp(timestamp metav1.Time) string {
-	if timestamp.IsZero() {
+func translateElapsedTimeUntilNow(timestamp metav1.Time) string {
+	return translateElapsedTime(metav1.Now(), timestamp)
+}
+
+// translateElapsedTime returns the elapsed time between the base time and
+// the target time in human-readable approximation.
+func translateElapsedTime(timestamp metav1.Time, base metav1.Time) string {
+	if timestamp.IsZero() || base.IsZero() {
 		return "<unknown>"
 	}
 
-	return duration.ShortHumanDuration(time.Since(timestamp.Time))
+	return duration.ShortHumanDuration(timestamp.Time.Sub(base.Time))
```

Comment:
You don't need the `base`; just copy what `translateTimestamp` does and use `time.Until` instead of `time.Since`. 

You'd probably want to rename `translateTimestamp` too as it's confusing (the misuse in pod describer proves it). For example, name them `timeSince` and `timeUntil` respectively.